src/core/models.py

```diff
@@ -863,19 +863,25 @@ def has_missing_image_files(self):
                 # attempt to pull a URL from the specified attribute
                 url = os.path.basename(val.get(attribute, None))
 
-                try:
-                    try:
-                        self.images.get(original_filename=url)
-                    except File.MultipleObjectsReturned:
-                        self.images.filter(original_filename=url).first()
-                except File.DoesNotExist:
+                if show_all:
                     missing_elements.append(url)
+                else:
+                    try:
+                        try:
+                            self.images.get(original_filename=url)
+                        except File.MultipleObjectsReturned:
+                            self.images.filter(original_filename=url).first()
```

Comment:
This looks confusing, should we just call `self.images.filter(original_filename=url).first()` since it already does what `self.images.get(original_filename=url)` would do?

Maybe worth adding a comment explaining why there might be more than one and why we are fine getting the first result